.maintain/Dockerfile

```diff
@@ -1,59 +1,55 @@
-# Note: We don't use Alpine and its packaged Rust/Cargo because they're too often out of date,
-# preventing them from being used to build Substrate/Polkadot.
-
-FROM phusion/baseimage:0.11 as builder
-LABEL maintainer="chevdor@gmail.com"
-LABEL description="This is the build stage for Substrate. Here we create the binary."
+FROM rust:1.46 as builder
+LABEL description="This is the build stage for Substrate-IPFS."
 
 ENV DEBIAN_FRONTEND=noninteractive
 
-ARG PROFILE=release
-WORKDIR /substrate
+WORKDIR /substrate-ipfs
+
+RUN apt update && \
+	apt install -y cmake pkg-config libssl-dev build-essential clang libclang-dev
+
+RUN rustup toolchain install nightly && \
+	rustup target add wasm32-unknown-unknown --toolchain nightly && \
+	rustup default stable
 
-COPY . /substrate
+COPY . /substrate-ipfs
 
+RUN	cargo build --release
+
+# Default target
+FROM debian:buster-slim
+LABEL co.equilibrium.image.vendor="Equilibrium Co." \
+	co.equilibrium.image.title="substrate-ipfs" \
+	co.equilibrium.image.description="IPFS-powered Substrate chain"
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# install tools and dependencies
 RUN apt-get update && \
-	apt-get dist-upgrade -y -o Dpkg::Options::="--force-confold" && \
-	apt-get install -y cmake pkg-config libssl-dev git clang
+	apt-get upgrade -y && \
+	apt-get install -y \
+		libssl1.1 \
+		ca-certificates && \
+# apt cleanup
+	apt-get autoremove -y && \
+	apt-get clean
+
+# add user (uid 1000; create group ipfs; w/ specified shell and home dir )
+RUN useradd -m -u 1000 -U -s /bin/sh -d /substrate-ipfs ipfs
+
+# add binaries to docker image
+COPY --from=builder /substrate-ipfs/target/release/substrate /usr/local/bin
+COPY --from=builder /substrate-ipfs/target/release/node-template /usr/local/bin
+
+# change user from root
+USER ipfs
+
+# check if executables work in this container
+RUN /usr/local/bin/substrate --version
+RUN /usr/local/bin/node-template --version
 
-RUN curl https://sh.rustup.rs -sSf | sh -s -- -y && \
-	export PATH="$PATH:$HOME/.cargo/bin" && \
-	rustup toolchain install nightly && \
-	rustup target add wasm32-unknown-unknown --toolchain nightly && \
-	rustup default stable && \
-	cargo build "--$PROFILE"
-
-# ===== SECOND STAGE ======
-
-FROM phusion/baseimage:0.11
-LABEL maintainer="chevdor@gmail.com"
-LABEL description="This is the 2nd stage: a very small image where we copy the Substrate binary."
-ARG PROFILE=release
-
-RUN mv /usr/share/ca* /tmp && \
-	rm -rf /usr/share/*  && \
-	mv /tmp/ca-certificates /usr/share/ && \
-	useradd -m -u 1000 -U -s /bin/sh -d /substrate substrate && \
-	mkdir -p /substrate/.local/share/substrate && \
-	chown -R substrate:substrate /substrate/.local && \
-	ln -s /substrate/.local/share/substrate /data
-
-COPY --from=builder /substrate/target/$PROFILE/substrate /usr/local/bin
-COPY --from=builder /substrate/target/$PROFILE/subkey /usr/local/bin
-COPY --from=builder /substrate/target/$PROFILE/node-rpc-client /usr/local/bin
```

Comment:
How big do these these binaries end up being? wondering if we should keep them around just in case.

Reply:
around 30MB in total, will see if can make the current image smaller to make room

Reply:
around 30MB in total, will see if can make the current image smaller to make room